chia/wallet/puzzles/singleton_truths.clib

```diff
@@ -0,0 +1,21 @@
+(
+  (defun-inline truth_data_to_truth_struct (my_id full_puzhash innerpuzhash my_amount lineage_proof singleton_struct) (c (c my_id  full_puzhash) (c (c innerpuzhash my_amount) (c lineage_proof singleton_struct))))
+
+  (defun-inline my_id_truth (Truths) (f (f Truths)))
+  (defun-inline my_full_puzzle_hash_truth (Truths) (r (f Truths)))
+  (defun-inline my_inner_puzzle_hash_truth (Truths) (f (f (r Truths))))
+  (defun-inline my_amount_truth (Truths) (r (f (r Truths))))
+  (defun-inline my_lineage_proof_truth (Truths) (f (r (r Truths))))
+  (defun-inline singleton_struct_truth (Truths) (r (r (r Truths))))
+
+  (defun-inline singleton_mod_hash_truth (Truths) (f (singleton_struct_truth Truths)))
+  (defun-inline singleton_launcher_id_truth (Truths) (f (r (singleton_struct_truth Truths))))
+  (defun-inline singleton_launcher_puzzle_hash_truth (Truths) (f (r (r (singleton_struct_truth Truths)))))
+
+  (defun-inline parent_info_for_lineage_proof (lineage_proof) (f lineage_proof))
+  (defun-inline puzzle_hash_for_lineage_proof (lineage_proof) (f (r lineage_proof)))
+  (defun-inline amount_for_lineage_proof (lineage_proof) (f (r (r lineage_proof))))
+  (defun-inline is_eve_proof (lineage_proof) (r (r lineage_proof)))
```

Comment:
I'm being a little nit-picky here, but this doesn't really have anything to do with singleton truths.

Reply:
(No need to fix this one.)

Reply:
My thinking was that it kinda does, because it's specific to a singleton lineage proof.  Different puzzles may have different versions of the lineage proof.  Plus it is also part of the Truths struct.